editor/src/components/common/server-hooks.ts

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react'
+import { isLocal } from '../editor/persistence'
+import { reduxDevtoolsLogMessage } from '../../core/shared/redux-devtools'
+import { checkProjectOwnership, fetchProjectMetadata } from '../../common/server'
+
+type ProjectMetadata = {
+  title: string | null
+  ownerName: string | null
+  ownerPicture: string | null
+}
+
+export function useGetProjectMetadata(projectId: string | null): ProjectMetadata {
+  const previousProjectIdRef = React.useRef<string | null>(null)
+  const [userData, setUserData] = React.useState<ProjectMetadata>({
+    title: null,
+    ownerName: null,
+    ownerPicture: null,
+  })
+  if (previousProjectIdRef.current !== projectId) {
+    previousProjectIdRef.current = projectId
+    if (projectId == null) {
+      setUserData({
+        title: null,
+        ownerName: null,
+        ownerPicture: null,
+      })
+    } else {
+      fetchProjectMetadata(projectId).then((projectListing) => {
+        // safeguard against an old Fetch arriving for an outdated projectId
+        if (previousProjectIdRef.current === projectId) {
+          setUserData({
+            title: projectListing?.title ?? null,
+            ownerName: projectListing?.ownerName ?? null,
+            ownerPicture: projectListing?.ownerPicture ?? null,
+          })
+        }
+      })
+    }
+  }
+
+  return userData
+}
+
+export function useIsMyProject(projectId: string | null): boolean {
+  const previousProjectIdRef = React.useRef<string | null>(null)
+  const [myProject, setMyProject] = React.useState<boolean>(false)
```

Comment:
Similarly here, it surely isn't that it's not the user's project, the answer to that isn't currently known until some time later.